### DISCUSSION TITLE
Computations on RPI

### DISCUSSION THREAD

**Discussion by slesinger** [frntc/Sidekick64#44] [Q&A] [answered]
Hi, would it be possible to bank-in a memory from an emulated cartridge and simultaneously to task Raspberry to update the memory? A C64 program would be able to use data computed by Raspberry.

I am also thinking about Raspberry to do networking via shared memory.

Thank you

**Comment by frntc** [comment, marked as answer]
Yes and no :) The challenging part is to reliably communicate with the C64 within the timing constraints of the bus. On the SK64 that works in an interrupt handler, i.e. in principle you can do other things meanwhile and also bank in memory. The closest to this in the repo is the menu code, and at least a video of some 3D rendering code is in the repo. What I never tried is multicore on the RPi, which might offer more possibilities and cause more problems ;)

Networking imposes additional problems: the network device is a USB-device which (at least in Circle) requires another interrupt which conflicts with the fast interrupt for the bus. To avoid problems, you could implement some protocol for testing "is the RPi available for communication". Something like this appears at various locations in the code.
